packages/gatsby/src/joi-schemas/joi.js

```diff
@@ -2,24 +2,58 @@ const Joi = require(`joi`)
 
 const stripTrailingSlash = chain => chain.replace(/(\w)\/+$/, `$1`)
 
-export const gatsbyConfigSchema = Joi.object().keys({
-  __experimentalThemes: Joi.array(),
-  polyfill: Joi.boolean(),
-  siteMetadata: Joi.object({
-    siteUrl: stripTrailingSlash(Joi.string()).uri(),
-  }).unknown(),
-  pathPrefix: Joi.string().uri({
-    allowRelative: true,
-    relativeOnly: true,
-  }),
-  mapping: Joi.object(),
-  plugins: Joi.array(),
-  proxy: Joi.object().keys({
-    prefix: Joi.string().required(),
-    url: Joi.string().required(),
-  }),
-  developMiddleware: Joi.func(),
-})
+export const gatsbyConfigSchema = Joi.object()
+  .keys({
+    __experimentalThemes: Joi.array(),
+    polyfill: Joi.boolean(),
+    assetPrefix: stripTrailingSlash(
+      Joi.string().uri({
+        allowRelative: true,
+      })
+    ),
+    pathPrefix: stripTrailingSlash(
+      Joi.string().uri({
+        allowRelative: true,
+        relativeOnly: true,
+      })
+    ),
+    siteMetadata: Joi.object({
+      siteUrl: stripTrailingSlash(Joi.string()).uri(),
+    }).unknown(),
+    mapping: Joi.object(),
+    plugins: Joi.array(),
+    proxy: Joi.object().keys({
+      prefix: Joi.string().required(),
+      url: Joi.string().required(),
+    }),
+    developMiddleware: Joi.func(),
+  })
+  // throws when both assetPrefix and pathPrefix are defined
+  .when(
```

Comment:
This is actually pretty cool! Gives us some conditional validation for gatsby-config.js. In this case, this will throw when assetPrefix and pathPrefix are both defined and both relative.